File: first_package/src/minimal_demo.cpp
#include <rclcpp/rclcpp.hpp>

int main(int argc, char* argv[]) {
    rclcpp::init(argc, argv);
    auto node = rclcpp::Node::make_shared("minimal_cpp");
    RCLCPP_ERROR_STREAM(node->get_logger(), "Hello from " << node->get_name());
    rclcpp::shutdown();
    return 0;
}

File: first_package/script/minimal_demo.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node


def main(args=None):
    rclpy.init(args=args)
    node = Node("minimal_py")
    node.get_logger().info(f"Hello from {node.get_name()}")
    rclpy.shutdown()


if __name__ == '__main__':
    main()


# If executables are deleted from lib/..
    # it will not run